### PR TITLE
Python 2 and 3 compatibility: Using basestring for unicode support

### PR DESCRIPTION
**Pull request by Kronuz**
Compatibility with Python 2 was incomplete (It was missing handling Unicode). This PR fixes it.

### CODE REVIEW

File: boolean/boolean.py [hunk 1]
```diff
@@ -12,6 +12,13 @@
 import itertools
 import collections
 
+# Python 3
```
Comment (pombredanne) [inline boolean/boolean.py]: Alternative would not a `from __future__ import unicode_literals` make this not needed?
Also what is the `# NOQA` about?

**Comment by Kronuz** [comment]
That wouldn't work, __future__ only affects inline strings, variables already containing Unicode won't be instances of `str` in Python 2.

`# NOQA` is just to avoid linting warnings

**Comment by bastikr** [comment]
I agree with the @Kronuz but can we instead use 
```python
try:
  basestring
except NameError:
  basestring = str
```
Is there a reason why you also redefined unicode?

**Comment by Kronuz** [comment]
@bastikr, it's just that it's a common way for adding Python 2 compatibility. It usually is an assignment because that snippet usually lives in some other "compat" module, and you then import from it (without the assignments you can't import the names). And Unicode was there only for completeness... but in our case your suggestion is fine, we can totally use that instead.

**Comment by Kronuz** [comment]
@bastikr updated the PR after your suggestion

**Comment by bastikr** [comment]
Thanks, I'll merge.

**Pull request merged**